packages/api/src/base/Decorate.ts

```diff
@@ -455,22 +454,15 @@ export default abstract class Decorate<ApiType extends ApiTypes> extends Events
           .subscribeStorage<[[Codec, Linkage<Codec>]]>([[creator, ...args]])
           .pipe(map(([data]): [Codec, Linkage<Codec>] => data))
         : this._rpcCore.state
-          .subscribeStorage<[LinkageResult]>([creator.iterKey])
+          .subscribeStorage<[LinkageResult]>([creator.keyPrefix()])
```

Comment:
```suggestion
          .subscribeStorage<[LinkageResult]>([creator.iterKey()])
```

The reason is the `iterKey` is meant to be the iteration key for maps explicitly. (Whereas `keyPrefix` does also exist for non-map types)